packages/labs/ssr-dom-shim/src/index.ts

```diff
@@ -61,6 +61,17 @@ const ElementShim = class Element {
   removeAttribute(name: string) {
     attributesForElement(this).delete(name);
   }
+  toggleAttribute(name: string, force?: boolean): boolean {
+    if (force === undefined) {
+      force = !this.hasAttribute(name);
+    }
+    if (force) {
+      this.setAttribute(name, '');
+    } else {
+      this.removeAttribute(name);
+    }
+    return force;
```

Comment:
This isn't quite correct, since if an attribute is already set and `force` is `true` the attribute should be left unmodified: https://dom.spec.whatwg.org/#dom-element-toggleattribute This code sets it to `''`.

I think this should work, but it needs a test for that case:

```suggestion
    if (this.hasAttribute(name) {
      // Step 5
      if (force === undefined || !force) {      
        this.removeAttribute(name);
        return false;
      }
    } else {
      // Step 4
      if (force === undefined || force) {
        // Step 4.1
        this.setAttribute(name, '');  
        return true;
      } else {
        // Step 4.2
        return false;
      }
    }
    // Step 6
    return true;
```

I _think_ this handles boolean conversion correctly. ie, `undefined` is the real missing value and `null` treated as falsey.

Reply:
Exceptional catch! Thank you! Fixed in https://github.com/lit/lit/pull/4494